orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -205,13 +204,13 @@ void addObjectMethodsForTraining(py::module& m) {
         return py::none();
       });
 
-  py::class_<onnxruntime::training::TrainingSession, InferenceSession> training_session(m, "TrainingSession");
-  training_session.def(py::init([](const SessionOptions& so) {
-                    Environment& env = get_env();
+  py::class_<onnxruntime::training::TrainingSession> training_session(m, "TrainingSession");
```

Comment:
Removed the inheritance from InferenceSession. AFAIK - The binding class need not inherit from InferenceSession as the original class TrainingSession inherits from InferenceSession anyway. This inheritance seems to complicate matters now that the inference python session is not InferenceSession (but rather PyInferenceSession) and I removed the inheritance as it doesn't seem necessary for the reason stated above.